### ISSUE TITLE
what's relationship between beta and temperature? 

### ISSUE THREAD

**Issue by kongya** [joshr17/HCL#8]
This paper claims that beta can tune the level of hardness: 
> The level of “hardness” in our method can be smoothly adjusted allowing the user to select the hardness that best trades-off between an improved learning signal from hard negatives, and the harm due to the correction of false negatives being only approximate. 

In paper {Understanding the Behaviour of Contrastive Loss} , the authors claim that: 
> We will show that the contrastive loss is a
hardness-aware loss function, and the temperature τ controls the strength of penalties on hard negative samples

So my question is : Are these two hyperparameters equivalent？

**Comment by joshr17** [comment]
There are two differences:

- `temperature` adjusts the weight for both positive and negative pairs. `beta` is applied only to the negative pairs, and so disentangles the weighting of positive and negative pairs.
- unlike  `temperature`, the hard negative sampling with  `beta` involves a normalization term across the batch of negatives. This means that, unlike with `temperature` there is a "budget allocation" occurring, where attributing more weight to one hard pair necessitates lower weighting on other negative pairs.
